containerize/containerize.sh

```diff
@@ -0,0 +1,195 @@
+	echo "-x [XRDIR]          the path to the tarball on an XRootD accessible device (default = ${XRDIR})"
+    echo "-h                  display this message and exit"
+	echo
+	echo "Examples:"
+	echo "./containerize.sh -t <tag> -b docker://docker.io/cmscloud/cc7-cms"
+	echo "podman run --rm -it -v /cvmfs/cms.cern.ch/:/cvmfs/cms.cern.ch/:ro 4a73e8eb3bf5f2a3ab84e2eedfb314e3489529e2b02a464cd7eeb0d2782091ff"
+	echo "./containerize.sh -t <tag> -b docker://docker.io/aperloff/cms-cvmfs-docker:light -C -v"
+	echo "podman run --rm -it <tag>"
+
+    exit ${EXIT}
+}
+
+# process options
+while getopts "b:cCd:f:j:t:T:vx:h" opt; do
+    case "$opt" in
+	b) BASE=$OPTARG
+	;;
+    c) KEEPCACHE=""
+    ;;
+	C) CLEAN="true"
+    ;;
+    d) DIR=$OPTARG
+    ;;
+    f) FILE=$OPTARG
+    ;;
+    j) JSON=$OPTARG
+    ;;
+    t) TAG=$OPTARG
+    ;;
+    T) TAR=$OPTARG
+    ;;
+	v) VCS=""
+    ;;
+    x) XRDIR=$OPTARG
+    ;;
+    h) usage 0
+    ;;
+    esac
+done
+
+dependency_check() {
+	subuid=$(cat /etc/subuid | grep "^`id -u`:")
+	subgid=$(cat /etc/subgid | grep "^`id -u`:")
+	if [ ! command -v buildah &> /dev/null ]; then
+		EXIT=$?
+		echo "Buildah could not be found!"
+		exit ${EXIT}
+	elif [[ -z "${subuid}" ]] || [[ -z "${subgid}" ]]; then
+		echo "Unable to find a subuid or subgid for id=`id -u` in /etc/subuid and /etc/subgid."
+		echo "Contact user support or your sysadmin for further assistance."
+	elif [[ ! -d /cvmfs/cms.cern.ch ]] || [ ! cvmfs_config probe cms.cern.ch &> /dev/null ]; then
+		EXIT=$?
+		echo "/cvmfs/cms.cern.ch must be mounted on the host to proceed."
+		exit ${EXIT}
+	fi
+
+	if [ ! command -v podman &> /dev/null ]; then
+		echo "Podman could not be found. While this is not strictly necessary, you will not be able to create a container from the resulting image."
+	fi
+}
+dependency_check
+
+# cache unneeded files
+CACHEDIR="""\
+Signature: 8a477f597d28d172789f06886806bc55\n
+# This file is a cache directory tag.\n
+# For information about cache directory tags, see:\n
+#       http://www.brynosaurus.com/cachedir/
+"""
+IFS=$'\n'
+list_of_cache_files=()
+# https://stackoverflow.com/questions/1955505/parsing-json-with-unix-tools
+for dir in $(jq -r '.Directories[] | .Path + " " + (.Cache|tostring)' ${JSON}); do
+	IFS=' '
+	dirarray=($dir)
+	path=${dirarray[0]}
+	path=`eval echo ${path}`
+	cache=${dirarray[1]}
+	cache_file=${path}/CACHEDIR.TAG
+	if [[ "${cache}" == "1" ]] && [[ ! -f ${cache_file} ]]; then
+		echo -e "Cache ${path}"
+		echo ${CACHEDIR} > ${cache_file}
+		list_of_cache_files=(${list_of_cache_files[@]} ${cache_file})
+	elif [[ "${cache}" == "1" ]] && [[ -f ${cache_file} ]]; then
+		echo -e "Already cached ${path}"
+	elif [[ "${cache}" == "0" ]] && [[ -f ${cache_file} ]]; then
+		echo -e "Uncache ${path}"
+		rm ${cache_file}
+	elif [[ "${cache}" == "0" ]] && [[ ! -f ${cache_file} ]]; then
+		echo "Already uncached ${path}"
+	fi
+done
+
+# tarball of CMSSW area
+if [ -z "${TAR}" ]; then
```

Comment:
would it be worth adding an option to force update a tarball that already exists?

Reply:
As far as I can tell if a tarball already exists it will simply be overwritten by the new one. I don't think there is a need for an extra flag.

Reply:
Yes, never mind, this comment did not make sense.